baseapp/baseapp_test.go

```diff
@@ -448,8 +450,41 @@ func TestTxGasLimits(t *testing.T) {
 	header := abci.Header{AppHash: []byte("apphash")}
 
 	app.BeginBlock(abci.RequestBeginBlock{Header: header})
-	res := app.Deliver(tx)
-	assert.Equal(t, res.Code, sdk.ToABCICode(sdk.CodespaceRoot, sdk.CodeOutOfGas), "Expected transaction to run out of gas")
+	resCheck := app.Check(tx)
+	assert.Equal(t, resCheck.Code, sdk.ToABCICode(sdk.CodespaceRoot, sdk.CodeOK), "Expected abci Codetype == 0, for CodeOK")
+	resDeliver := app.Deliver(tx)
+	assert.Equal(t, resDeliver.Code, sdk.ToABCICode(sdk.CodespaceRoot, sdk.CodeOutOfGas), "Expected transaction to run out of gas due to fee in msg handler")
+	app.EndBlock(abci.RequestEndBlock{})
+	app.Commit()
+}
+
+//Ante handler uses gas, and should trigger a failure for CheckTx() and DeliverTx(). DeliverTx()'s msg handler should never run
+func TestFeeFailureCheckTx(t *testing.T) {
```

Comment:
As I mentioned in an above comment - we should also have a similar test to this which checks multiple (valid) checkTx statements in a row - where there is only enough gas for the first one (which should pass) where as all subsequent checkTx's should fail